Core/clim-core/presentations/presentation-type-functions.lisp

```diff
@@ -263,29 +279,42 @@ otherwise return false."
            (return-from presentation-subtypep (values yp yp))))
         ((and (eq maybe-subtype-name 'completion)
               (not (eq maybe-supertype-name 'completion)))
-         (destructuring-bind (sequence
-                              &key (value-key 'identity)
-                              &allow-other-keys)
-             maybe-subtype-parameters
-           (return-from presentation-subtypep
-             (values
-              (every (lambda (elt)
-                       (presentation-typep (funcall value-key elt) maybe-supertype))
-                     sequence)
-              t))))
+         (return-from presentation-subtypep
+           (values (every-completion-item
+                    (alexandria:rcurry #'presentation-typep maybe-supertype)
+                    maybe-subtype-parameters)
+                   t)))
+        ((and (not (eq maybe-subtype-name 'completion))
+              (eq maybe-supertype-name 'completion))
+         (return-from presentation-subtypep
+           (values t nil)))
         ((and (eq maybe-subtype-name 'subset-completion)
               (eq maybe-supertype-name 'sequence))
-         (destructuring-bind (sequence
-                              &key (value-key 'identity)
-                              &allow-other-keys)
-             maybe-subtype-parameters
-           (let ((element-type (first maybe-supertype-parameters)))
-             (return-from presentation-subtypep
-               (values
-                (every (lambda (elt)
-                         (presentation-typep (funcall value-key elt) element-type))
-                       sequence)
-                t)))))))
+         (let ((element-type (first maybe-supertype-parameters)))
+           (return-from presentation-subtypep
+             (values (every-completion-item
+                      (alexandria:rcurry #'presentation-typep element-type)
+                      maybe-subtype-parameters)
+                     t))))
+        ((and (eq maybe-subtype-name 'sequence-enumerated)
+              (eq maybe-supertype-name 'sequence))
+         (let ((sub-element-types maybe-subtype-parameters)
+               (super-element-type (first maybe-supertype-parameters)))
+           (return-from presentation-subtypep
+             (values (every (alexandria:rcurry #'presentation-subtypep
+                                               super-element-type)
+                            sub-element-types)
+                     t))))
+        ((and (eq maybe-subtype-name 'sequence-enumerated)
+              (eq maybe-supertype-name 'sequence-enumerated))
```

Comment:
this case should be left for the presentation method presentatin-subtypep

Reply:
Yeah, that works.